src/linearAlgebra.jl

```diff
@@ -8,7 +8,28 @@ function norm(pv::PartitionedVector, p::Real = 2; kwargs...)
   return _norm
 end
 
-function dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}; kwargs...) where {T}
+dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}; kwargs...) where {T} = dot(pv1, pv2, Val(pv1.simulate_vector), Val(pv2.simulate_vector); kwargs...)
+
+function dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{true}, ::Val{false}; kwargs...) where {T}
```

Comment:
I think it's worth adding a short docstring here. I am giving an example, that would just miss a sentence on the `::Val{true}, ::Val{false}` thing.

```suggestion
"""
    dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{true}, ::Val{false}; kwargs...)

Compute the dot product between two partitioned vectors.
"""
function dot(pv1::PartitionedVector{T}, pv2::PartitionedVector{T}, ::Val{true}, ::Val{false}; kwargs...) where {T}
```